Automattic/jetpack
# Create plugin support for CDN'ing of static assets.

Add in CDN hosted versions of Core / Jetpack scripts and styles to speed up sites and better cross-site cacheability.

#### To do:

- [x] Add caching to external http calls so they're not requested on each load.
- [x] Abstract out plugin script/style swapping to enable for other plugins (like WooCommerce) as well.
- [x] Confirm RTL styles are handled correctly.

#### Testing instructions:

* Activate Photon CDN through the old module settings. Confirm that scripts / styles are now pointing to c0.wp.com resources in the page source. If testing using a development version, you can use a filter to override versions to production versions:
```
add_filter( 'jetpack_cdn_plugin_slug_and_version', function( $values ) {
	return array( 'jetpack', '6.5' );
} );
add_filter( 'jetpack_cdn_core_version_and_locale', function( $values ) {
	return array( '4.9.7', 'en_US' );
} );
```
* Make sure everything loads in a couple browsers.
* Test in multisite
* Test in subfolder installs.
* Test in RTL

#### Proposed changelog entry for your changes:

* Speed up sites and increase max concurrent connections through Photon by cloud-hosting Jetpack and WordPress Core scripts, styles, and assets in addition to content images.
